core/src/com/unciv/ui/newgamescreen/NewGameScreen.kt

```diff
@@ -38,6 +39,12 @@ class NewGameScreen(
     private val playerPickerTable: PlayerPickerTable
     private val mapOptionsTable: MapOptionsTable
 
+    @Throws(InterruptedException::class, IOException::class)
+    fun isConnected(): Boolean {
+        val command = "ping -c 1 github.com"
```

Comment:
I'm all for Wind-hose bashing, but I still know -c is a routing compartment on Woodledoze boxes, so that will fail unless the process has admin and elevation. And github is not the most reliable round-the-clock icmp partner. And does that even start on non-rooted Android? Even if, pinging is highest permissions on droid too.

Reply:
I tested it on a non-root galaxy A52S and it worked with no problem
You're right about github.com, maybe use 8.8.8.8 instead? (the Google DNS)
I was thinking of making it ping 2 sites and consider it offline only if both fail... maybe 8.8.8.8 and 1.1.1.1

Reply:
New, on Droid there are platform-specific API's for exactly this usecase and those should be used. And as I said - fails on Win:vomiting_face: , and W-can't say it - is still a supported platform.

Reply:
> Naw, on Droid there are platform-specific API's for exactly this usecase and those should be used. And as I said - fails on Win🤮 , and W-can't say it - is still a supported platform.

ohhh I see, I'll try to fix that.
Would checking for the OS be a good way to fix this? for android it will run -c and for windows -n

Reply:
For starters, yes. Detection is already somewhere in the codebase, look for.... CrashReportSysInfoDesktop. That one can also be used to learn how platform-specific stuff is handled around here - I mean how it is called and where it lives and so on.
And droid should start at: https://developer.android.com/reference/android/net/ConnectivityManager

Reply:
I'm running into issues. I try to import android.net.ConnectivityManager and it gives me "unresolved refrence: net". Even if I try android.* it says "unresolved refrence: android". I do have the right sdk and sdk tools installed and after multiple tries of invalidating caches and resyncing gradle files it still doesn't work. Do you happen to have an idea of whats going on here?

Reply:
See linked PR - import is no issue at all within the android module.